BMSSender.Test/BatteryMeasureStreamTest.cs

```diff
@@ -0,0 +1,62 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using BMSSender;
+
+namespace BMSSender.Test
+{
+    [TestClass]
+    public class BatteryMeasureStreamTest
+    {
+        [TestMethod]
+        public void GivenReadingCount_WhenInvalidInput_ReturnsFalse()
+        {
+            bool _isValidReadingCount = UserInputValidation.IsNumeric("test");
+            Assert.IsFalse(_isValidReadingCount);
+        }
+
+        [TestMethod]
+        public void GivenReadingCount_WhenValidInput_ReturnsTrue()
+        {
+            bool _isValidReadingCount = UserInputValidation.IsNumeric("15");
+            Assert.IsTrue(_isValidReadingCount);
+        }
+
+        [TestMethod]
+        public void GivenReadingCount_WhenLesserThan15_ReturnsFalse()
+        {
+            bool _isValidReadingCount = UserInputValidation.IsGreaterThanOrEqualTo15("2");
+            Assert.IsFalse(_isValidReadingCount);
+        }
+
+        [TestMethod]
+        public void GivenReadingCount_WhenGreaterOrEquals15_ReturnsTrue()
+        {
+            bool _isValidReadingCount = UserInputValidation.IsGreaterThanOrEqualTo15("20");
+            Assert.IsTrue(_isValidReadingCount);
+        }
+
+        [TestMethod]
+        public void GivenReadingCount_WhenValidCount_CreatesReadingList()
+        {
+            BatteryMeasureStream measures = new BatteryMeasureStream();
+            measures.CreateBatteryMeasureReadingsList(15);
+            Assert.IsTrue(measures._readings.Count >= 15);
+        }
+
+        [TestMethod]
+        public void GenerateTemperatureValue_ReturnsDoubleTypeValueWithinExtremeLimits()
```

Comment:
These tests are checking random number generation. They may pass even when the ranges are wrong. Focus on testing your functionality - the most crucial is the output format.
In the absence of a test for the output format, any future mistakes will be caught only while integrating with the receiver.